CodeStarterCamp_Week3/main.swift

```diff
@@ -14,52 +14,77 @@ struct Person {
     var cardBalance: Int
     
     
-    mutating func orderMenu(coffeePrice: Int) {
-        if cardBalance >= coffeePrice {
-            cardBalance -= coffeePrice
-            print("주문이 완료되었습니다.")
-        }
-        else {
-            print("잔액 부족으로 주문이 취소되었습니다.")
+    mutating func order(_ coffee: Coffee, of coffeeShop: CoffeeShop, by name: String) {
+        print("손님, 원하시는 음료 \(coffee.rawValue)가 맞으시나요?")
+        
+        if let orderCoffeePrice = coffeeShop.coffeeMenu[coffee] {
+            print("결제 도와드리겠습니다.")
+            print("주문하신 \(coffee.rawValue)의 가격은 \(orderCoffeePrice)원 입니다.")
+            
+            if cardBalance >= orderCoffeePrice {
+                self.cardBalance -= orderCoffeePrice
+                coffeeShop.totalRevenue += orderCoffeePrice
+                
+                coffeeShop.receipt(coffee, self, orderCoffeePrice)
+                
+                coffeeShop.make(coffee, for: name)
+            }
+            else {
+                print("잔액이 \(orderCoffeePrice - cardBalance)원만큼 부족합니다.")
+            }
         }
     }
 }
 
-enum Coffee {
-    case americano, cappuccino, latte, espresso, mocha
+enum Coffee: String{
+    case americano = "아메리카노"
+    case cappuccino = "카푸치노"
+    case latte = "라떼"
+    case espresso = "에스프레소"
+    case mocha = "모카"
 }
 
-struct CoffeeShop {
+class CoffeeShop {
     var coffeeShopName: String
     var totalRevenue: Int = 0
     var coffeeMenu: [Coffee: Int] = [.americano: 4500, .cappuccino: 5000, .espresso: 4000, .latte: 5000, .mocha: 5500]
-    var pickUpTable: [Coffee] = []
+    var pickUpTable: [Coffee.RawValue: String] = [:]
```

Comment:
여기를 프로퍼티 옵저버로 구현하는 게 좋아보여요.
간단합니다. `didSet`을 활용하시면 돼요!

```suggestion
    var pickUpTable: [Coffee.RawValue: String] {
        didSet {
            print("\(pickUpTable.values) 님이 주문하신 \(pickUpTable.keys)(이/가) 준비되었습니다. 픽업대에서 가져가주세요.")
        }
    }
```

제가 조언드린 코드로는 딕셔너리의 배열 형태로 출력이 되는데요. 
이 부분은 개미님이 String 타입으로 바꿔보시길 바랍니다!